README.md

```diff
@@ -1,5 +1,7 @@
 # coreaudio-rs [![Build Status](https://travis-ci.org/RustAudio/coreaudio-rs.svg?branch=master)](https://travis-ci.org/RustAudio/coreaudio-rs) [![Crates.io](https://img.shields.io/crates/v/coreaudio-rs.svg)](https://crates.io/crates/coreaudio-rs) [![Crates.io](https://img.shields.io/crates/l/coreaudio-rs.svg)](https://github.com/RustAudio/coreaudio-rs/blob/master/LICENSE-MIT)
 
+ADDED: This is a fork of the original coreaudio-rs whose only difference with the original is that audio streams are not forced to have the static lifetime.  Will be removed from this account if and when our PR is implemented.
+
```

Comment:
This note should be removed if we're going to merge this into master.

Reply:
My own reason is that I'm writing an audio DSP library in Rust that's meant to run in core, so Arc is out of the question and I use `Cell<T>`s to share signals between inputs.  One of the several reason for this choice is that my code is meant to run as fast as possible, and on small processors.  I can send you the current state of my work if you're interested, although I learned from @andrewcsmith that some of the things I believe about Sample are not true anymore, and my writeup should be updated, at least before I post in on GitHub.

I'm not the only one who wants non-static streams for audio (and MIDI and OSC) backends, for example [this thread](https://github.com/RustAudio/rust-portaudio/issues/151) .

Reply:
Ah yes I remember this issue, as well as suggesting the possiblity of adding a lifetime param to the portaudio Stream.

There are two other possibilities that come to mind that might allow us to remove the `'static` requirement from the callback without needing to add a lifetime to `Stream`, though I'm not sure how feasible either are:

1. Perhaps there's some way of using the "self-borrowing" feature of the [rental](https://docs.rs/rental/0.4.14/rental/) crate to solve the lifetime issue and remove the need for adding a lifetime param to the Stream? I'm not sure how hackish this would be though as I haven't personally used rental, but I've heard some folks have found it useful.
2. Maybe we could do something similar to [scoped_threadpool](https://github.com/Kimundi/scoped-threadpool-rs) which adds an extra scope to use for restricting the lifetime.

I think it's worth exploring this before adding a param to the stream itself as this creates other ergonomic issues. I definitely agree it's a problem worth solving though, as there really should not be a need for `Arc` in this case. Ideally we'd get some kind of non-borrowing lifetime solution from the language itself, but i'm not sure if there is an RFC for anything like this yet.

>  I can send you the current state of my work if you're interested

Sure, I'd be more than happy to take a look at the stuff that uses the `sample` crate! The crate is still under fairly heavy development at the moment so any feedback is invaluable. I'd be curious to hear what kind of pitfalls you've run into with it, and I'm particularly curious about what you mean by using `Cell` to share inputs? Do you mean in order to avoid the need for `Bus`? I've actually been working on a similar  Also it sounds like you're possibly considering using it on an embedded project? Would love to hear more if you're willing/able to share :)

Reply:
This is just a quick reply.  I remember looking at `rental` and not finding it useful for my needs.  But that was a long time ago and I was (still am to quite a degree) a Rust beginner.  Didn't know about `scoped_threadpool` and will look at it.

Your `sample` crate was the inspiration for my work, and I owe a lot to it, but I decided to start from scratch.  Some of the the reasons for this move are articulated in the documentation for my own library.

I'm sending you the current state of things.  Not much has been implemented, but I spent quite some time working on the general shape of the API and perfecting best practices.

Reply:
I think I found a way not to need a static lifetime on my audio streams, but I would still argue that this change would make the library more useful and versatile.  Do you have examples where it would really break things?  Or maybe you could include it as an option via a `#[cfg(...)]` thingie?